pkg/plugin/plugin.go

```diff
@@ -133,7 +152,37 @@ func (r *RpcPlugin) SetWeight(rollout *v1alpha1.Rollout, desiredWeight int32, ad
 }
 
 func (r *RpcPlugin) SetHeaderRoute(rollout *v1alpha1.Rollout, headerRouting *v1alpha1.SetHeaderRoute) pluginTypes.RpcError {
-	return pluginTypes.RpcError{}
+	r.LogCtx.Debugln("SetHeaderRoute")
+	ctx := context.TODO()
+
+	glooPluginConfig, err := getPluginConfig(rollout)
+	if err != nil {
+		return pluginTypes.RpcError{
+			ErrorString: err.Error(),
+		}
+	}
+	if glooPluginConfig.CanaryDestination == nil {
+		return pluginTypes.RpcError{
+			ErrorString: "CanaryDestination must be specified when using setHeaderRoute with " + PluginName,
+		}
+	}
+
+	// get the matched routetables
+	matchedRts, err := r.getRouteTables(ctx, rollout, glooPluginConfig)
+	if err != nil {
+		return pluginTypes.RpcError{
+			ErrorString: err.Error(),
+		}
+	}
+	if len(matchedRts) == 0 {
+		// nothing to update, don't bother computing things
+		return pluginTypes.RpcError{}
```

Comment:
Should we throw an error if nothing is selected? The GP API is not entirely consistent in handling this scenario today, but we've definitely had feedback that it's confusing UX for users if e.g. selectors don't match because of a missing label or typo, they're wondering why the entire resource "does nothing".

Reply:
What would this error say?

"Unable to find RouteTables which match {labelSelectors}, a matching RouteTable must already exist because the glooplatform plugin does not create new RouteTables"

Reply:
I think just the first part is probably sufficient:

> Unable to find any RouteTables which match {labelSelectors}

It at least gives folks a starting point for troubleshooting. If we have enough flexibility for this to be a "warning" rather than an "error", that's even better... but IDK if we have that flexibility with this particular API

Reply:
You can select a rt by name, name/namespace or labelSelectors so for now I added a basic error with a TODO to enhance it with the search criteria used which failed to produce a satisfactory result